Index: audacity-src-1.3.5-beta/lib-src/portaudio-v19/Makefile.in
===================================================================
--- audacity-src-1.3.5-beta.orig/lib-src/portaudio-v19/Makefile.in
+++ audacity-src-1.3.5-beta/lib-src/portaudio-v19/Makefile.in
@@ -39,7 +39,7 @@ OTHER_OBJS = @OTHER_OBJS@
 PALIB = libportaudio.la
 PAINC = include/portaudio.h
 
-PA_LDFLAGS = $(LDFLAGS) $(SHARED_FLAGS) -rpath $(libdir) -no-undefined \
+PA_LDFLAGS = $(LDFLAGS) -rpath $(libdir) -no-undefined \
 	     -export-symbols-regex "(Pa|PaMacCore|PaJack|PaAlsa)_.*" \
 	     -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
