diff --git a/lib-src/portaudio-v19/src/os/unix/pa_unix_util.c b/lib-src/portaudio-v19/src/os/unix/pa_unix_util.c
index 18f806c..7f4fd540 100644
--- a/lib-src/portaudio-v19/src/os/unix/pa_unix_util.c
+++ b/lib-src/portaudio-v19/src/os/unix/pa_unix_util.c
@@ -154,6 +154,10 @@ void PaUtil_InitializeClock( void )
 }
 
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 PaTime PaUtil_GetTime( void )
 {
 #ifdef HAVE_MACH_ABSOLUTE_TIME
@@ -169,6 +173,10 @@ PaTime PaUtil_GetTime( void )
 #endif
 }
 
+#ifdef __cplusplus
+}
+#endif
+
 PaError PaUtil_InitializeThreading( PaUtilThreading *threading )
 {
     (void) paUtilErr_;

