diff --git a/lib-src/portaudio-v19/src/hostapi/wmme/pa_win_wmme.c b/lib-src/portaudio-v19/src/hostapi/wmme/pa_win_wmme.c
index 422c86714..9047149ff 100644
--- a/lib-src/portaudio-v19/src/hostapi/wmme/pa_win_wmme.c
+++ b/lib-src/portaudio-v19/src/hostapi/wmme/pa_win_wmme.c
@@ -1751,7 +1751,11 @@ static PaError CalculateBufferSettings(
 
                 if( *hostFramesPerOutputBuffer != *hostFramesPerInputBuffer )
                 {
-                    if( hostFramesPerInputBuffer < hostFramesPerOutputBuffer )
+// JKC: Patched By Audacity.  Our Bug 1969
+// Previously this line incorrectly read:
+// if( hostFramesPerInputBuffer < hostFramesPerOutputBuffer )
+// So it was comparing pointers, rather than the values pointed to.
+                    if( *hostFramesPerInputBuffer < *hostFramesPerOutputBuffer )
                     {
                         *hostFramesPerOutputBuffer = *hostFramesPerInputBuffer;
 
