diff --git a/lib-src/portaudio-v19/src/common/pa_debugprint.c b/lib-src/portaudio-v19/src/common/pa_debugprint.c
index f34d4bbf9..7e9e628fe 100644
--- a/lib-src/portaudio-v19/src/common/pa_debugprint.c
+++ b/lib-src/portaudio-v19/src/common/pa_debugprint.c
@@ -71,7 +71,7 @@ void PaUtil_SetDebugPrintFunction(PaUtilLogCallback cb)
 }
 
 /*
- If your platform doesn’t have vsnprintf, you are stuck with a
+ If your platform doesn't have vsnprintf, you are stuck with a
  VERY dangerous alternative, vsprintf (with no n)
 */
 #if _MSC_VER

