Index: lib-src/portaudio-v19/Makefile.in
===================================================================
--- lib-src/portaudio-v19/Makefile.in	(revision 13510)
+++ lib-src/portaudio-v19/Makefile.in	(working copy)
@@ -156,8 +156,11 @@
 SUBDIRS =
 @ENABLE_CXX_TRUE@SUBDIRS += bindings/cpp
 
-all: lib/$(PALIB) all-recursive tests examples selftests
+prep:
+	mkdir -p qa/loopback/src/ src/hostapi/skeleton/ $(SRC_DIRS)
 
+all: prep lib/$(PALIB) all-recursive tests examples selftests
+
 tests: bin-stamp $(TESTS)
 
 examples: bin-stamp $(EXAMPLES)
