diff --git a/lib-src/portaudio-v19/Makefile.in b/lib-src/portaudio-v19/Makefile.in
index 22ed79ad3..98b3680eb 100644
--- a/lib-src/portaudio-v19/Makefile.in
+++ b/lib-src/portaudio-v19/Makefile.in
@@ -56,6 +56,7 @@ COMMON_OBJS = \
 	src/common/pa_dynload.o \
 	src/common/pa_front.o \
 	src/common/pa_process.o \
+	src/common/pa_ringbuffer.o \
 	src/common/pa_stream.o \
 	src/common/pa_trace.o \
 	src/hostapi/skeleton/pa_hostapi_skeleton.o
