diff --git a/lib-src/portaudio-v19/src/hostapi/jack/pa_jack.c b/lib-src/portaudio-v19/src/hostapi/jack/pa_jack.c
index 887128c3b..787d81ddb 100644
--- a/lib-src/portaudio-v19/src/hostapi/jack/pa_jack.c
+++ b/lib-src/portaudio-v19/src/hostapi/jack/pa_jack.c
@@ -441,7 +441,7 @@ static void PaJack_WaitForStreamDataToBecomeAvailable( PaJackStream *stream )
 #endif
 }
 
-#if defined(WIN32)
+#if defined(WIN32) && _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
 
diff --git a/lib-src/portaudio-v19/src/hostapi/wdmks/pa_win_wdmks.c b/lib-src/portaudio-v19/src/hostapi/wdmks/pa_win_wdmks.c
index 70a48d8ff..d413391a2 100644
--- a/lib-src/portaudio-v19/src/hostapi/wdmks/pa_win_wdmks.c
+++ b/lib-src/portaudio-v19/src/hostapi/wdmks/pa_win_wdmks.c
@@ -101,7 +101,7 @@ of a device for the duration of active stream using those devices
 
 #include <math.h>
 
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && _MSC_VER < 1900
 #define snprintf _snprintf
 #define vsnprintf _vsnprintf
 #endif
