Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12938)
+++ Makefile.am	(working copy)
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I M4
+
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
 if BUILD_OCTAVE_MOD
Index: configure.ac
===================================================================
--- configure.ac	(revision 12938)
+++ configure.ac	(working copy)
@@ -15,16 +15,13 @@
 AC_CONFIG_MACRO_DIR([M4])
 AC_CONFIG_HEADERS([src/config.h])
 
-AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
+AM_INIT_AUTOMAKE
 AM_SILENT_RULES([yes])
 
 dnl Audacity policy: don't enable automatic rebuild of configure et al if 
 dnl sources change
 AM_MAINTAINER_MODE([disable])
 
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
-
 AC_LANG([C])
 
 AC_PROG_CC
Index: examples/Makefile.am
===================================================================
--- examples/Makefile.am	(revision 12938)
+++ examples/Makefile.am	(working copy)
@@ -2,7 +2,7 @@
 
 noinst_PROGRAMS = make_sine sfprocess list_formats generate sndfilehandle sndfile-to-text
 
-INCLUDES = -I$(top_srcdir)/src
+AM_CPPFLAGS = -I$(top_srcdir)/src
 
 sndfile_to_text_SOURCES = sndfile-to-text.c
 sndfile_to_text_LDADD = $(top_builddir)/src/libsndfile.la
Index: Octave/Makefile.am
===================================================================
--- Octave/Makefile.am	(revision 12938)
+++ Octave/Makefile.am	(working copy)
@@ -21,7 +21,7 @@
 OCT_LIBS = @OCT_LIBS@
 
 SNDFILEDIR = $(top_builddir)/src
-INCLUDES = -I$(SNDFILEDIR)
+AM_CPPFLAGS = -I$(SNDFILEDIR)
 
 oct_module_srcs = sndfile.cc
 oct_module_files = sndfile.oct PKG_ADD
@@ -39,10 +39,10 @@
 # Use Octave's mkoctfile to do all the heavy lifting. Unfortunately, its
 # a little dumb so we need to guide it carefully.
 sndfile.oct : sndfile.o
-	$(QUIET_GEN) $(MKOCTFILE) -v $(INCLUDES) $(top_builddir)/Octave/$+ -L$(SNDFILEDIR)/.libs -L$(SNDFILEDIR) -lsndfile -o $(top_builddir)/Octave/$@ > /dev/null
+	$(QUIET_GEN) $(MKOCTFILE) -v $(AM_CPPFLAGS) $(top_builddir)/Octave/$+ -L$(SNDFILEDIR)/.libs -L$(SNDFILEDIR) -lsndfile -o $(top_builddir)/Octave/$@ > /dev/null
 
 sndfile.o : sndfile.cc
-	$(QUIET_GEN) $(MKOCTFILE) -v $(INCLUDES) -c $+ -o $(top_builddir)/Octave/$@ > /dev/null
+	$(QUIET_GEN) $(MKOCTFILE) -v $(AM_CPPFLAGS) -c $+ -o $(top_builddir)/Octave/$@ > /dev/null
 
 # Allow for the test being run in the build dir, but the test script
 # being located in the source dir.
Index: programs/Makefile.am
===================================================================
--- programs/Makefile.am	(revision 12938)
+++ programs/Makefile.am	(working copy)
@@ -7,7 +7,7 @@
 OS_SPECIFIC_CFLAGS = @OS_SPECIFIC_CFLAGS@
 OS_SPECIFIC_LINKS = @OS_SPECIFIC_LINKS@
 
-INCLUDES = -I$(top_srcdir)/src $(OS_SPECIFIC_CFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir)/src $(OS_SPECIFIC_CFLAGS)
 
 CLEANFILES = *~
 
Index: regtest/Makefile.am
===================================================================
--- regtest/Makefile.am	(revision 12938)
+++ regtest/Makefile.am	(working copy)
@@ -4,7 +4,7 @@
 
 noinst_HEADERS = regtest.h
 
-INCLUDES = -I$(top_srcdir)/src $(SQLITE3_CFLAGS) $(OS_SPECIFIC_CFLAGS)
+AM_CPPFLAGS = -I$(top_srcdir)/src $(SQLITE3_CFLAGS) $(OS_SPECIFIC_CFLAGS)
 
 sndfile_regtest_SOURCES = sndfile-regtest.c database.c checksum.c
 sndfile_regtest_LDADD = $(top_builddir)/src/libsndfile.la $(SQLITE3_LIBS)
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 12938)
+++ src/Makefile.am	(working copy)
@@ -2,7 +2,7 @@
 
 SUBDIRS = GSM610 G72x
 
-INCLUDES = @EXTERNAL_CFLAGS@
+AM_CPPFLAGS = @EXTERNAL_CFLAGS@
 
 lib_LTLIBRARIES = libsndfile.la
 include_HEADERS = sndfile.hh
Index: tests/Makefile.am
===================================================================
--- tests/Makefile.am	(revision 12938)
+++ tests/Makefile.am	(working copy)
@@ -6,7 +6,7 @@
 CPP_TEST = cpp_test
 endif
 
-INCLUDES = -I$(top_srcdir)/src
+AM_CPPFLAGS = -I$(top_srcdir)/src
 
 noinst_PROGRAMS = sfversion floating_point_test write_read_test \
 	lossy_comp_test error_test ulaw_test alaw_test dwvw_test \
