Index: libsndfile/src/aiff.c
===================================================================
--- libsndfile/src/aiff.c	(revision 10749)
+++ libsndfile/src/aiff.c	(working copy)
@@ -519,7 +519,8 @@
 						psf->datalength -= ssnd_fmt.offset ;
 						}
 					else
-					{	psf_log_printf (psf, "  Offset     : %u (Should be zero)\n", ssnd_fmt.offset) ;
+					{	psf->dataoffset += ssnd_fmt.offset ;
+						psf_log_printf (psf, "  Offset     : %u (Should be zero)\n", ssnd_fmt.offset) ;
 						psf_log_printf (psf, "  Block Size : %u ???\n", ssnd_fmt.blocksize) ;
 						} ;
 
