Index: lib-src/libsndfile/Makefile.am
===================================================================
--- lib-src/libsndfile/Makefile.am	(revision 13510)
+++ lib-src/libsndfile/Makefile.am	(working copy)
@@ -8,7 +8,7 @@
 octave_dir = Octave
 endif
 
-SUBDIRS = M4 man doc Win32 src $(octave_dir) examples regtest tests programs
+SUBDIRS = M4 man doc Win32 src #$(octave_dir) examples regtest tests programs
 DIST_SUBDIRS = M4 man doc Win32 src Octave examples regtest tests programs
 
 EXTRA_DIST = libsndfile.spec.in sndfile.pc.in
Index: lib-src/libsndfile/Makefile.in
===================================================================
--- lib-src/libsndfile/Makefile.in	(revision 13510)
+++ lib-src/libsndfile/Makefile.in	(working copy)
@@ -412,7 +412,7 @@
 ACLOCAL_AMFLAGS = -I M4
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 @BUILD_OCTAVE_MOD_TRUE@octave_dir = Octave
-SUBDIRS = M4 man doc Win32 src $(octave_dir) examples regtest tests programs
+SUBDIRS = M4 man doc Win32 src #$(octave_dir) examples regtest tests programs
 DIST_SUBDIRS = M4 man doc Win32 src Octave examples regtest tests programs
 EXTRA_DIST = libsndfile.spec.in sndfile.pc.in
 CLEANFILES = *~
Index: lib-src/libsndfile/configure
===================================================================
--- lib-src/libsndfile/configure	(revision 13510)
+++ lib-src/libsndfile/configure	(working copy)
@@ -18986,11 +18986,10 @@
 $as_echo "$as_me: WARNING: ******************************************************************" >&2;}
 	fi
 
-if test x"$ac_cv_prog_HAVE_AUTOGEN" = "xno" ; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Touching files in directory tests/." >&5
-$as_echo "$as_me: WARNING: Touching files in directory tests/." >&2;}
-	touch tests/*.c tests/*.h
-	fi
+#if test x"$ac_cv_prog_HAVE_AUTOGEN" = "xno" ; then
+#	AC_MSG_WARN([[Touching files in directory tests/.]])
+#	touch tests/*.c tests/*.h
+#	fi
 
 #====================================================================================
 # Settings for the HTML documentation.
@@ -22427,5 +22426,5 @@
 	echo
 	fi
 
-(cd src && make genfiles)
-(cd tests && make genfiles)
+#(cd src && make genfiles)
+#(cd tests && make genfiles)
Index: lib-src/libsndfile/configure.ac
===================================================================
--- lib-src/libsndfile/configure.ac	(revision 13510)
+++ lib-src/libsndfile/configure.ac	(working copy)
@@ -457,10 +457,10 @@
 	AC_MSG_WARN([[******************************************************************]])
 	fi
 
-if test x"$ac_cv_prog_HAVE_AUTOGEN" = "xno" ; then
-	AC_MSG_WARN([[Touching files in directory tests/.]])
-	touch tests/*.c tests/*.h
-	fi
+#if test x"$ac_cv_prog_HAVE_AUTOGEN" = "xno" ; then
+#	AC_MSG_WARN([[Touching files in directory tests/.]])
+#	touch tests/*.c tests/*.h
+#	fi
 
 #====================================================================================
 # Settings for the HTML documentation.
@@ -691,5 +691,5 @@
 	echo
 	fi
 
-(cd src && make genfiles)
-(cd tests && make genfiles)
+#(cd src && make genfiles)
+#(cd tests && make genfiles)
