--- ../../../../libsndfile-1.0.18pre25/M4/endian.m4	2008-01-16 10:51:17.000000000 +0000
+++ M4/endian.m4	2009-02-08 17:30:40.000000000 +0000
@@ -135,7 +135,6 @@
 	fi
 
 )
-]
 
 if test $ac_cv_c_byte_order = big ; then
 	ac_cv_c_big_endian=1
@@ -154,6 +153,6 @@
 	AC_MSG_WARN([[*****************************************************************]])
 	fi
 
-)# AC_C_FIND_ENDIAN
+])# AC_C_FIND_ENDIAN
 
 
