--- configure.ac.orig	2011-03-23 04:01:02.000000000 -0500
+++ configure.ac	2011-04-02 00:50:13.000000000 -0500
@@ -18,6 +18,10 @@
 AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
 AM_SILENT_RULES([yes])
 
+dnl Audacity policy: don't enable automatic rebuild of configure et al if 
+dnl sources change
+AM_MAINTAINER_MODE([enable])
+
 dnl Add parameters for aclocal
 AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
 
