Index: src/common.c
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/src/common.c,v
retrieving revision 1.11
diff -u -r1.11 src/common.c
--- src/common.c	8 Feb 2009 18:44:57 -0000	1.11
+++ src/common.c	9 Feb 2009 03:58:15 -0000
@@ -23,7 +23,9 @@
 #include	<ctype.h>
 #include	<math.h>
 #include	<time.h>
+#if defined(HAVE_SYS_TIME_H)
 #include	<sys/time.h>
+#endif
 
 #include	"sndfile.h"
 #include	"sfendian.h"
Index: src/mpc2k.c
===================================================================
RCS file: /cvsroot/audacity/lib-src/libsndfile/src/mpc2k.c,v
retrieving revision 1.1
diff -u -r1.1 src/mpc2k.c
--- src/mpc2k.c	8 Feb 2009 18:44:57 -0000	1.1
+++ src/mpc2k.c	9 Feb 2009 04:14:07 -0000
@@ -134,7 +134,7 @@
 	if (psf->is_pipe == SF_FALSE)
 		psf_fseek (psf, 0, SEEK_SET) ;
 
-	snprintf (sample_name, sizeof (sample_name), "%s                    ", psf->filename) ;
+	LSF_SNPRINTF (sample_name, sizeof (sample_name), "%s                    ", psf->filename) ;
 
 	psf_binheader_writef (psf, "e11b", 1, 4, sample_name, make_size_t (HEADER_NAME_LEN)) ;
 	psf_binheader_writef (psf, "e111", 100, 0, (psf->sf.channels - 1) & 1) ;

