Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12764)
+++ Makefile.am	(working copy)
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
+
 #AUTOMAKE_OPTIONS = 1.7 foreign dist-zip dist-bzip2
 AUTOMAKE_OPTIONS = foreign 1.11 dist-zip dist-xz
 
Index: acinclude.m4
===================================================================
--- acinclude.m4	(revision 12764)
+++ acinclude.m4	(working copy)
@@ -121,8 +121,8 @@
      echo "*** to the full path to pkg-config."
      echo "*** Or see http://www.freedesktop.org/software/pkgconfig to get pkg-config."
   else
-     PKG_CONFIG_MIN_VERSION=0.9.0
-     if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
+     pkg_config_min_version=0.9.0
+     if $PKG_CONFIG --atleast-pkgconfig-version $pkg_config_min_version; then
         AC_MSG_CHECKING(for $2)
 
         if $PKG_CONFIG --exists "$2" ; then
@@ -148,7 +148,7 @@
         AC_SUBST($1_CFLAGS)
         AC_SUBST($1_LIBS)
      else
-        echo "*** Your version of pkg-config is too old. You need version $PKG_CONFIG_MIN_VERSION or newer."
+        echo "*** Your version of pkg-config is too old. You need version $pkg_config_min_version or newer."
         echo "*** See http://www.freedesktop.org/software/pkgconfig"
      fi
   fi
Index: configure.ac
===================================================================
--- configure.ac	(revision 12764)
+++ configure.ac	(working copy)
@@ -11,13 +11,11 @@
 
 AC_CANONICAL_TARGET([])
 
-AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
+AM_INIT_AUTOMAKE
 AM_MAINTAINER_MODE
 AM_CONFIG_HEADER([config.h])
+AC_CONFIG_MACRO_DIR([m4])
 
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I m4")
-
 dnl Library versioning
 dnl - library source changed -> increment REVISION
 dnl - interfaces added/removed/changed -> increment CURRENT, REVISION = 0
Index: examples/Makefile.am
===================================================================
--- examples/Makefile.am	(revision 12764)
+++ examples/Makefile.am	(working copy)
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-INCLUDES = -I$(top_srcdir)/include @OGG_CFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/include @OGG_CFLAGS@
 
 noinst_PROGRAMS = decoder_example encoder_example chaining_example\
 		vorbisfile_example seeking_example
Index: lib/Makefile.am
===================================================================
--- lib/Makefile.am	(revision 12764)
+++ lib/Makefile.am	(working copy)
@@ -2,7 +2,7 @@
 
 SUBDIRS = modes books
 
-INCLUDES = -I$(top_srcdir)/include @OGG_CFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/include @OGG_CFLAGS@
 
 lib_LTLIBRARIES = libvorbis.la libvorbisfile.la libvorbisenc.la
 
Index: test/Makefile.am
===================================================================
--- test/Makefile.am	(revision 12764)
+++ test/Makefile.am	(working copy)
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-INCLUDES = -I$(top_srcdir)/include @OGG_CFLAGS@
+AM_CPPFLAGS = -I$(top_srcdir)/include @OGG_CFLAGS@
 
 check_PROGRAMS = test
 
Index: vq/Makefile.am
===================================================================
--- vq/Makefile.am	(revision 12764)
+++ vq/Makefile.am	(working copy)
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I../lib -I$(top_srcdir)/include @OGG_CFLAGS@
+AM_CPPFLAGS = -I../lib -I$(top_srcdir)/include @OGG_CFLAGS@
 
 EXTRA_PROGRAMS = latticebuild latticetune huffbuild distribution
 CLEANFILES = $(EXTRA_PROGRAMS)
