Index: lib-src/libflac/src/libFLAC/ia32/nasm.h
===================================================================
RCS file: /cvsroot/audacity/lib-src/libflac/src/libFLAC/ia32/nasm.h,v
retrieving revision 1.2
diff -u -r1.2 nasm.h
--- lib-src/libflac/src/libFLAC/ia32/nasm.h	13 Feb 2006 11:36:45 -0000	1.2
+++ lib-src/libflac/src/libFLAC/ia32/nasm.h	12 Jan 2008 17:26:55 -0000
@@ -49,6 +49,11 @@
 	%idefine code_section section .text align=16
 	%idefine data_section section .data align=32
 	%idefine bss_section  section .bss  align=32
+%elifdef OBJ_FORMAT_macho
+	%define FLAC__PUBLIC_NEEDS_UNDERSCORE
+	%idefine code_section section .text
+	%idefine data_section section .data
+	%idefine bss_section  section .bss
 %else
 	%error unsupported object format!
 %endif
