--- Makefile.am~	2007-06-22 04:51:56.000000000 +0100
+++ Makefile.am	2008-02-02 14:38:08.000000000 +0000
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = 1.6 foreign dist-zip dist-bzip2
 
-SUBDIRS = lib include doc examples vq
+SUBDIRS = lib include vq
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = vorbis.m4
--- configure.in~	2008-02-02 14:30:32.000000000 +0000
+++ configure.in	2008-02-02 15:18:27.000000000 +0000
@@ -257,9 +257,7 @@
 lib/books/coupled/Makefile
 lib/books/uncoupled/Makefile
 lib/books/floor/Makefile
-doc/Makefile doc/vorbisfile/Makefile doc/vorbisenc/Makefile
 include/Makefile include/vorbis/Makefile
-examples/Makefile
 vq/Makefile
 libvorbis.spec
 vorbis.pc
