diff -wruN orig/wxMac-2.8.11/src/unix/utilsunx.cpp wxMac-2.8.11/src/unix/utilsunx.cpp
--- orig/wxMac-2.8.11/src/unix/utilsunx.cpp	2010-04-15 04:25:58.000000000 -0500
+++ wxMac-2.8.11/src/unix/utilsunx.cpp	2011-03-30 00:07:28.000000000 -0500
@@ -1202,8 +1202,8 @@
         wxMAC_MachPortEndProcessDetect helper function.  Though in practice
         this shouldn't be a problem since it wasn't prototyped anywhere.
  */
-#define USE_OLD_DARWIN_END_PROCESS_DETECT (defined(__DARWIN__) && defined(__WXMAC__))
-// #define USE_OLD_DARWIN_END_PROCESS_DETECT 0
+// #define USE_OLD_DARWIN_END_PROCESS_DETECT (defined(__DARWIN__) && defined(__WXMAC__))
+#define USE_OLD_DARWIN_END_PROCESS_DETECT 0
 
 // wxMac doesn't use the same process end detection mechanisms so we don't
 // need wxExecute-related helpers for it.
