Index: lib-src/libid3tag/Makefile.am
===================================================================
--- lib-src/libid3tag/Makefile.am	(revision 13510)
+++ lib-src/libid3tag/Makefile.am	(working copy)
@@ -95,27 +95,27 @@
 
 libid3tag_la_LDFLAGS =	-version-info $(version_info)
 
-BUILT_SOURCES =		frametype.c compat.c genre.dat
+#BUILT_SOURCES =		frametype.c compat.c genre.dat
+#
+#$(srcdir)/frametype.c: $(srcdir)/frametype.gperf Makefile.am
+#	cd $(srcdir) &&  \
+#	gperf -tCcTonD -K id -N id3_frametype_lookup -s -3 -k '*'  \
+#		frametype.gperf |  \
+#	sed -e 's/\(struct id3_frametype\);/\1/' |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >frametype.c
+#
+#$(srcdir)/compat.c: $(srcdir)/compat.gperf Makefile.am
+#	cd $(srcdir) &&  \
+#	gperf -tCcTonD -K id -N id3_compat_lookup -s -3 -k '*'  \
+#		compat.gperf |  \
+#	sed -e 's/\(struct id3_compat\);/\1/' |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >compat.c
+#
+#$(srcdir)/genre.dat: $(srcdir)/genre.dat.in $(srcdir)/genre.dat.sed Makefile.am
+#	cd $(srcdir) &&  \
+#	sed -n -f genre.dat.sed genre.dat.in |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >genre.dat
 
-$(srcdir)/frametype.c: $(srcdir)/frametype.gperf Makefile.am
-	cd $(srcdir) &&  \
-	gperf -tCcTonD -K id -N id3_frametype_lookup -s -3 -k '*'  \
-		frametype.gperf |  \
-	sed -e 's/\(struct id3_frametype\);/\1/' |  \
-	sed -e '/\$$''Id: /s/\$$//g' >frametype.c
-
-$(srcdir)/compat.c: $(srcdir)/compat.gperf Makefile.am
-	cd $(srcdir) &&  \
-	gperf -tCcTonD -K id -N id3_compat_lookup -s -3 -k '*'  \
-		compat.gperf |  \
-	sed -e 's/\(struct id3_compat\);/\1/' |  \
-	sed -e '/\$$''Id: /s/\$$//g' >compat.c
-
-$(srcdir)/genre.dat: $(srcdir)/genre.dat.in $(srcdir)/genre.dat.sed Makefile.am
-	cd $(srcdir) &&  \
-	sed -n -f genre.dat.sed genre.dat.in |  \
-	sed -e '/\$$''Id: /s/\$$//g' >genre.dat
-
 libtool: $(LIBTOOL_DEPS)
 	$(SHELL) ./config.status --recheck
 
Index: lib-src/libid3tag/Makefile.in
===================================================================
--- lib-src/libid3tag/Makefile.in	(revision 13510)
+++ lib-src/libid3tag/Makefile.in	(working copy)
@@ -415,8 +415,7 @@
 			debug.c debug.h
 
 libid3tag_la_LDFLAGS = -version-info $(version_info)
-BUILT_SOURCES = frametype.c compat.c genre.dat
-all: $(BUILT_SOURCES) config.h
+all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
 .SUFFIXES:
@@ -874,8 +873,7 @@
 	       $(distcleancheck_listfiles) ; \
 	       exit 1; } >&2
 check-am: all-am
-check: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) check-recursive
+check: check-recursive
 all-am: Makefile $(LTLIBRARIES) $(HEADERS) config.h
 installdirs: installdirs-recursive
 installdirs-am:
@@ -882,8 +880,7 @@
 	for dir in "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
-install: $(BUILT_SOURCES)
-	$(MAKE) $(AM_MAKEFLAGS) install-recursive
+install: install-recursive
 install-exec: install-exec-recursive
 install-data: install-data-recursive
 uninstall: uninstall-recursive
@@ -913,7 +910,6 @@
 maintainer-clean-generic:
 	@echo "This command is intended for maintainers to use"
 	@echo "it deletes files that may require special tools to rebuild."
-	-test -z "$(BUILT_SOURCES)" || rm -f $(BUILT_SOURCES)
 clean: clean-recursive
 
 clean-am: clean-generic clean-libLTLIBRARIES clean-libtool \
@@ -988,8 +984,7 @@
 
 uninstall-am: uninstall-includeHEADERS uninstall-libLTLIBRARIES
 
-.MAKE: $(am__recursive_targets) all check install install-am \
-	install-strip
+.MAKE: $(am__recursive_targets) all install-am install-strip
 
 .PHONY: $(am__recursive_targets) CTAGS GTAGS TAGS all all-am \
 	am--refresh check check-am clean clean-cscope clean-generic \
@@ -1011,25 +1006,27 @@
 	uninstall-includeHEADERS uninstall-libLTLIBRARIES
 
 
-$(srcdir)/frametype.c: $(srcdir)/frametype.gperf Makefile.am
-	cd $(srcdir) &&  \
-	gperf -tCcTonD -K id -N id3_frametype_lookup -s -3 -k '*'  \
-		frametype.gperf |  \
-	sed -e 's/\(struct id3_frametype\);/\1/' |  \
-	sed -e '/\$$''Id: /s/\$$//g' >frametype.c
+#BUILT_SOURCES =		frametype.c compat.c genre.dat
+#
+#$(srcdir)/frametype.c: $(srcdir)/frametype.gperf Makefile.am
+#	cd $(srcdir) &&  \
+#	gperf -tCcTonD -K id -N id3_frametype_lookup -s -3 -k '*'  \
+#		frametype.gperf |  \
+#	sed -e 's/\(struct id3_frametype\);/\1/' |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >frametype.c
+#
+#$(srcdir)/compat.c: $(srcdir)/compat.gperf Makefile.am
+#	cd $(srcdir) &&  \
+#	gperf -tCcTonD -K id -N id3_compat_lookup -s -3 -k '*'  \
+#		compat.gperf |  \
+#	sed -e 's/\(struct id3_compat\);/\1/' |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >compat.c
+#
+#$(srcdir)/genre.dat: $(srcdir)/genre.dat.in $(srcdir)/genre.dat.sed Makefile.am
+#	cd $(srcdir) &&  \
+#	sed -n -f genre.dat.sed genre.dat.in |  \
+#	sed -e '/\$$''Id: /s/\$$//g' >genre.dat
 
-$(srcdir)/compat.c: $(srcdir)/compat.gperf Makefile.am
-	cd $(srcdir) &&  \
-	gperf -tCcTonD -K id -N id3_compat_lookup -s -3 -k '*'  \
-		compat.gperf |  \
-	sed -e 's/\(struct id3_compat\);/\1/' |  \
-	sed -e '/\$$''Id: /s/\$$//g' >compat.c
-
-$(srcdir)/genre.dat: $(srcdir)/genre.dat.in $(srcdir)/genre.dat.sed Makefile.am
-	cd $(srcdir) &&  \
-	sed -n -f genre.dat.sed genre.dat.in |  \
-	sed -e '/\$$''Id: /s/\$$//g' >genre.dat
-
 libtool: $(LIBTOOL_DEPS)
 	$(SHELL) ./config.status --recheck
 
