Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12902)
+++ Makefile.am	(working copy)
@@ -3,8 +3,15 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = sbsms.pc
 
-libtool: $(LIBTOOL_DEPS)
-	$(SHELL) ./config.status --recheck
-
 #tell aclocal that extra macros can be found in m4/
 ACLOCAL_AMFLAGS = -I m4
+
+EXTRA_DIST = \
+	build/macosx.sh \
+	win/config.h \
+	win/sbsms/sbsms.ICO \
+	win/sbsms/sbsms.rc \
+	win/sbsms/sbsms.sln \
+	win/sbsms/sbsms.vcproj \
+	win/sbsms/sbsms.wxs \
+	$(NULL)
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 12902)
+++ src/Makefile.am	(working copy)
@@ -4,4 +4,30 @@
 libsbsmsinclude_HEADERS = ../include/sbsms.h
 
 lib_LTLIBRARIES = libsbsms.la
-libsbsms_la_SOURCES = sms.cpp track.cpp trackpoint.cpp resample.cpp sbsms.cpp fft.cpp grain.cpp buffer.cpp dBTable.cpp slide.cpp dBTable.h synthTable.h sse.h subband.cpp buffer.h fft.h utils.h trackpoint.h grain.h real.h resample.h ../include/sbsms.h sms.h subband.h track.h
+libsbsms_la_SOURCES = \
+	sms.cpp \
+	track.cpp \
+	trackpoint.cpp \
+	resample.cpp \
+	sbsms.cpp \
+	fft.cpp \
+	grain.cpp \
+	buffer.cpp \
+	dBTable.cpp \
+	slide.cpp \
+	dBTable.h \
+	sincCoeffs.h \
+	synthTable.h \
+	sse.h \
+	subband.cpp \
+	buffer.h \
+	fft.h \
+	utils.h \
+	trackpoint.h \
+	grain.h \
+	real.h \
+	../include/sbsms.h \
+	sms.h \
+	subband.h \
+	track.h \
+	$(NULL)
