Index: configure.ac
===================================================================
--- configure.ac	(revision 12842)
+++ configure.ac	(working copy)
@@ -13,7 +13,7 @@
 LIBSBSMS_RELEASE_BUILD=0
 AC_CONFIG_SRCDIR([src/sbsms.cpp])
 AC_CANONICAL_TARGET([])
-AM_CONFIG_HEADER(src/config.h)
+AC_CONFIG_HEADERS([src/config.h])
 
 AC_DISABLE_SHARED dnl allows for optimizations
 
@@ -36,8 +36,8 @@
 
 SHARED_VERSION_INFO="10:0:0"
 
-AC_PROG_LIBTOOL
-AM_PROG_LIBTOOL
+AM_PROG_AR
+LT_INIT
 
 # save $CXXFLAGS etc. since AC_PROG_CXX likes to insert "-g -O2"
 # if $CXXFLAGS is blank and it finds GCC
@@ -65,7 +65,7 @@
 
 AC_ARG_ENABLE(multithreaded,[  --enable-multithreaded enable multithreaded build: (default: disable)],[enable_multithreaded=$enableval],[enable_multithreaded=no])
 
-AC_ARG_ENABLE(sse, [AC_HELP_STRING([--enable-sse],[enable SSE optimizations])], enable_sse=$enableval, enable_sse=yes)
+AC_ARG_ENABLE(sse, [AS_HELP_STRING([--enable-sse],[enable SSE optimizations])], enable_sse=$enableval, enable_sse=yes)
 
 AC_ARG_ENABLE(universal_binary,[  --enable-universal_binary enable universal binary build: (default: disable)],[enable_universal_binary=$enableval],[enable_universal_binary=no])
 
Index: m4/ac_c99_func_lrint.m4
===================================================================
--- m4/ac_c99_func_lrint.m4	(revision 12839)
+++ m4/ac_c99_func_lrint.m4	(working copy)
@@ -13,13 +13,12 @@
 AC_DEFUN([AC_C99_FUNC_LRINT],
 [AC_CACHE_CHECK(for lrint,
   ac_cv_c99_lrint,
-[AC_TRY_COMPILE([
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define		_ISOC9X_SOURCE	1
 #define 	_ISOC99_SOURCE	1
 #define		__USE_ISOC99	1
 #define 	__USE_ISOC9X	1
-#include	<math.h>],
-[	int value = lrint (0.432) ; ], ac_cv_c99_lrint=yes, ac_cv_c99_lrint=no)])
+#include	<math.h>]], [[	int value = lrint (0.432) ; ]])],[ac_cv_c99_lrint=yes],[ac_cv_c99_lrint=no])])
 if test "x$ac_cv_c99_lrint" = "xyes"; then
   AC_DEFINE(HAVE_LRINT, 1,
             [Define if you have C99's lrint function.])
Index: m4/ac_c99_func_lrintf.m4
===================================================================
--- m4/ac_c99_func_lrintf.m4	(revision 12839)
+++ m4/ac_c99_func_lrintf.m4	(working copy)
@@ -13,13 +13,12 @@
 AC_DEFUN([AC_C99_FUNC_LRINTF],
 [AC_CACHE_CHECK(for lrintf,
   ac_cv_c99_lrintf,
-[AC_TRY_COMPILE([
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define		_ISOC9X_SOURCE	1
 #define 	_ISOC99_SOURCE	1
 #define		__USE_ISOC99	1
 #define 	__USE_ISOC9X	1
-#include	<math.h>],
-[	int value = lrintf (0.432) ; ], ac_cv_c99_lrintf=yes, ac_cv_c99_lrintf=no)])
+#include	<math.h>]], [[	int value = lrintf (0.432) ; ]])],[ac_cv_c99_lrintf=yes],[ac_cv_c99_lrintf=no])])
 if test "x$ac_cv_c99_lrintf" = "xyes"; then
   AC_DEFINE(HAVE_LRINTF, 1,
             [Define if you have C99's lrintf function.])
