--- ../../libmad-0.15.1b/layer12.c	2004-02-05 09:02:39.000000000 +0000
+++ libmad/layer12.c	2006-10-19 11:44:19.000000000 +0100
@@ -16,7 +16,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  *
- * $Id: libmad-mp2-out-of-spec.patch,v 1.1 2008-02-01 20:29:07 richardash1981 Exp $
+ * $Id: libmad-mp2-out-of-spec.patch,v 1.1 2008-02-01 20:29:07 richardash1981 Exp $
  */
 
 # ifdef HAVE_CONFIG_H
@@ -363,6 +363,7 @@
 # endif
     }
     else {  /* nch == 1 */
+#if 0 /* relax restriction, those files do import fine! */
       if (bitrate_per_channel > 192000) {
 	/*
 	 * ISO/IEC 11172-3 does not allow single channel mode for 224, 256,
@@ -371,6 +372,7 @@
 	stream->error = MAD_ERROR_BADMODE;
 	return -1;
       }
+#endif
     }
 
     if (bitrate_per_channel <= 48000)
