Index: fixed.h
===================================================================
RCS file: /cvsroot/audacity/lib-src/libmad/fixed.h,v
retrieving revision 1.2
diff -u -r1.2 fixed.h
--- fixed.h	12 Sep 2005 07:01:30 -0000	1.2
+++ fixed.h	2 Feb 2008 11:35:08 -0000
@@ -137,7 +137,7 @@
 
 /* --- Intel --------------------------------------------------------------- */
 
-# elif defined(FPM_INTEL)
+# elif defined(FPM_INTEL) || defined (__i386__)
 
 #  if defined(_MSC_VER)
 #   pragma warning(push)
@@ -351,7 +351,7 @@
 
 /* --- PowerPC ------------------------------------------------------------- */
 
-# elif defined(FPM_PPC)
+# elif defined(FPM_PPC) || defined(__ppc__)
 
 /*
  * This PowerPC version is fast and accurate; the disposition of the least
