--- orig/include/wx/msw/setup.h	2014-10-06 14:33:44.000000000 -0700
+++ include/wx/msw/setup.h	2015-10-08 17:32:21.223729900 -0700
@@ -65,11 +65,11 @@
 // NDEBUG) or, on the contrary, enable more asserts, including the usually
 // disabled ones, in the debug build (then do it inside #ifndef NDEBUG)
 //
-// #ifdef NDEBUG
-//  #define wxDEBUG_LEVEL 0
-// #else
-//  #define wxDEBUG_LEVEL 2
-// #endif
+#ifdef NDEBUG
+ #define wxDEBUG_LEVEL 0
+#else
+ #define wxDEBUG_LEVEL 1
+#endif
 
 // wxHandleFatalExceptions() may be used to catch the program faults at run
 // time and, instead of terminating the program with a usual GPF message box,
@@ -1366,7 +1366,7 @@
 // Default is 0.
 //
 // Recommended setting (at present): 0
-#define wxUSE_ACCESSIBILITY 0
+#define wxUSE_ACCESSIBILITY 1
 
 // ----------------------------------------------------------------------------
 // miscellaneous settings
