diff -wruN orig/wxWidgets-3.0.2/include/wx/msw/setup.h wxWidgets-3.0.2/include/wx/msw/setup.h
--- orig/include/wx/msw/setup.h 2014-10-06 16:33:44.000000000 -0500
+++ include/wx/msw/setup.h      2014-10-14 18:38:37.992586000 -0500
@@ -1366,7 +1366,7 @@
 // Default is 0.
 //
 // Recommended setting (at present): 0
-#define wxUSE_ACCESSIBILITY 0
+#define wxUSE_ACCESSIBILITY 1

 // ----------------------------------------------------------------------------
 // miscellaneous settings
