Index: docs/changes.txt
===================================================================
--- docs/changes.txt	(revision 2632977f6b6aa2f5c6254a69936ca288ace0560d)
+++ docs/changes.txt	(revision 0797a6f8754db982b87a1df63975ccf76df2905f)
@@ -613,4 +613,5 @@
 
 - Fix compilation with g++ 4.9 in non-C++11 mode.
+- Fix regression in accessibility support (Leland Lucius).
 - Fix regression in wxDC drawing with bottom-to-top y axis (Artur Wieczorek).
 - Fix compilation with C++Builder XE compiler (Nichka).
Index: src/msw/window.cpp
===================================================================
--- src/msw/window.cpp	(revision 8bbe683f61af82bfb3025ccf4d23885d211ca90b)
+++ src/msw/window.cpp	(revision 0797a6f8754db982b87a1df63975ccf76df2905f)
@@ -3387,5 +3387,6 @@
                 if (dwObjId == (LPARAM)OBJID_CLIENT && GetOrCreateAccessible())
                 {
-                    return LresultFromObject(IID_IAccessible, wParam, (IUnknown*) GetAccessible()->GetIAccessible());
+                    processed = true;
+                    rc.result = LresultFromObject(IID_IAccessible, wParam, (IUnknown*) GetAccessible()->GetIAccessible());
                 }
                 break;
