diff -ur flac-1.2.1.orig/src/flac/Makefile.am flac-1.2.1/src/flac/Makefile.am
--- flac-1.2.1.orig/src/flac/Makefile.am	2007-09-11 20:56:51.000000000 +0300
+++ flac-1.2.1/src/flac/Makefile.am	2007-09-20 19:38:09.000000000 +0300
@@ -53,6 +53,6 @@
 	$(top_builddir)/src/share/utf8/libutf8.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	@OGG_LIBS@ \
-	@LIBICONV@ \
+	@LTLIBICONV@ \
 	@MINGW_WINSOCK_LIBS@ \
 	-lm
diff -ur flac-1.2.1.orig/src/metaflac/Makefile.am flac-1.2.1/src/metaflac/Makefile.am
--- flac-1.2.1.orig/src/metaflac/Makefile.am	2007-09-11 20:59:25.000000000 +0300
+++ flac-1.2.1/src/metaflac/Makefile.am	2007-09-20 19:38:25.000000000 +0300
@@ -49,6 +49,6 @@
 	$(top_builddir)/src/share/utf8/libutf8.la \
 	$(top_builddir)/src/libFLAC/libFLAC.la \
 	@OGG_LIBS@ \
-	@LIBICONV@ \
+	@LTLIBICONV@ \
 	@MINGW_WINSOCK_LIBS@ \
 	-lm
