Index: libflac/src/Makefile.in
===================================================================
--- libflac/src/Makefile.in	(revision 11513)
+++ libflac/src/Makefile.in	(working copy)
@@ -220,6 +220,7 @@
 
 SUBDIRS = \
 	libFLAC \
+	flac \
 	share \
 	$(CPPLIBS_DIRS)
 
@@ -240,7 +241,7 @@
 	installdirs-recursive install-recursive uninstall-recursive \
 	check-recursive installcheck-recursive
 DIST_COMMON = $(srcdir)/Makefile.in Makefile.am
-DIST_SUBDIRS = libFLAC share libFLAC++
+DIST_SUBDIRS = libFLAC share libFLAC++ flac
 all: all-recursive
 
 .SUFFIXES:
