Index: configure.ac
===================================================================
--- configure.ac	(revision 12834)
+++ configure.ac	(working copy)
@@ -21,6 +21,7 @@
 
 AC_INIT(SoundTouch, 1.7.0, [http://www.surina.net/soundtouch])
 AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_MACRO_DIR([config/m4])
 AM_CONFIG_HEADER([include/soundtouch_config.h])
 AM_INIT_AUTOMAKE
 #AC_DISABLE_SHARED	dnl This makes libtool only build static libs 
Index: config/am_include.mk
===================================================================
--- config/am_include.mk	(revision 12837)
+++ config/am_include.mk	(working copy)
@@ -27,7 +27,7 @@
 ## $(top_srcdir) macro is set by configure. It's important to use $(top_srcdir)
 ## in case a user decides to build in a separate directory from the base package
 ## directory. Using absolute, or relative paths is a bad idea.
-INCLUDES=-I$(top_srcdir)/include
+AM_CPPFLAGS=-I$(top_srcdir)/include
 
 
 # doc directory
