diff --git a/lib-src/libnyquist/nyquist/nyqsrc/sound.c b/lib-src/libnyquist/nyquist/nyqsrc/sound.c
index 35de780..d81beb5 100644
--- a/lib-src/libnyquist/nyquist/nyqsrc/sound.c
+++ b/lib-src/libnyquist/nyquist/nyqsrc/sound.c
@@ -1588,6 +1588,7 @@ double step_to_hz(double steps)
 }
 
 #ifdef WIN32
+#ifndef _MSC_VER < 1800
 #define RECIP_LOG_2 1.44269504088895364453
 
 double log2(double x)
@@ -1595,6 +1596,7 @@ double log2(double x)
   return log(x) * RECIP_LOG_2;
 }
 #endif
+#endif
 
 /*
  * from old stuff...
