--- lilv/src/orig_lilv_internal.h	2012-11-23 00:00:15.000000000 -0600
+++ lilv/src/lilv_internal.h	2013-10-27 05:18:03.970327900 -0500
@@ -30,7 +30,6 @@
 #    include <windows.h>
 #    define dlopen(path, flags) LoadLibrary(path)
 #    define dlclose(lib) FreeLibrary((HMODULE)lib)
-#    define dlsym GetProcAddress
 #    ifdef _MSC_VER
 #        define __func__ __FUNCTION__
 #        define INFINITY DBL_MAX + DBL_MAX
@@ -363,9 +362,13 @@
 static inline VoidFunc
 lilv_dlfunc(void* handle, const char* symbol)
 {
+#ifdef _WIN32
+   return (VoidFunc)GetProcAddress((HMODULE)handle, symbol);
+#else
 	typedef VoidFunc (*VoidFuncGetter)(void*, const char*);
 	VoidFuncGetter dlfunc = (VoidFuncGetter)dlsym;
 	return dlfunc(handle, symbol);
+#endif
 }
 
 #ifdef LILV_DYN_MANIFEST
