diff --git a/redland/configure.ac b/redland/configure.ac
index d7a41ce..0c8cf99 100644
--- a/redland/configure.ac
+++ b/redland/configure.ac
@@ -210,7 +210,7 @@ dnl Checks for libraries.
 
 
 dnl Berkeley DB
-AC_ARG_WITH(bdb, [  --with-bdb=DIR          Berkeley DB install area (default=/usr)], bdb_prefix="$withval", bdb_prefix="none") 
+AC_ARG_WITH(bdb, [  --with-bdb=DIR          Berkeley DB install area (default=/usr)], bdb_prefix="$withval", bdb_prefix="no") 
 AC_ARG_WITH(bdb_lib, [  --with-bdb-lib=DIR      Berkeley DB lib directory (default=/usr/lib)], bdb_lib_dir="$withval", bdb_lib_dir="none") 
 AC_ARG_WITH(bdb_include, [  --with-bdb-include=DIR  Berkeley DB include directory (default=/usr/include)], bdb_include_dir="$withval", bdb_include_dir="none") 
 AC_ARG_WITH(bdb_dbname, [  --with-bdb-dbname=NAME  Berkeley DB library name (auto)], bdb_dbname="$withval", bdb_dbname="none")
@@ -442,7 +442,7 @@ LIBS="$LIBRDF_LIBS"
 dnl Checks for XML parsers
 
 # At present this is for passing on to raptor and not used here
-AC_ARG_WITH(xml-parser, [  --with-xml-parser=NAME  Use XML parser - expat, libxml (default=libxml)], xml_parser="$withval", xml_parser="libxml") 
+AC_ARG_WITH(xml-parser, [  --with-xml-parser=NAME  Use XML parser - expat, libxml (default=libxml)], xml_parser="$withval", xml_parser="expat") 
 
 
 
@@ -453,7 +453,7 @@ rdf_parsers_available=
 
 # raptor is REQUIRED despite the checking here
 AC_CHECK_PROG(RAPTOR_CONFIG, raptor-config, raptor-config)
-AC_ARG_WITH(raptor, [  --with-raptor=SOURCE    Pick raptor source - system/internal (default=auto)], with_raptor="$withval", with_raptor="auto")
+AC_ARG_WITH(raptor, [  --with-raptor=SOURCE    Pick raptor source - system/internal (default=internal)], with_raptor="$withval", with_raptor="internal")
 
 RAPTOR_MIN_VERSION=1.4.17
 AC_SUBST(RAPTOR_MIN_VERSION)
@@ -515,7 +515,7 @@ fi
 
 # rasqal is REQUIRED despite the checking here
 AC_CHECK_PROG(RASQAL_CONFIG, rasqal-config, rasqal-config)
-AC_ARG_WITH(rasqal, [  --with-rasqal=SOURCE    Pick rasqal source - system/internal (default=auto)], with_rasqal="$withval", with_rasqal="auto")
+AC_ARG_WITH(rasqal, [  --with-rasqal=SOURCE    Pick rasqal source - system/internal (default=internal)], with_rasqal="$withval", with_rasqal="internal")
 
 RASQAL_MIN_VERSION=0.9.16
 RASQAL_MAX_VERSION=0.9.99
@@ -565,7 +565,7 @@ AC_MSG_RESULT($digest_modules)
 DIGEST_OBJS=
 DIGEST_SRCS=
 
-AC_ARG_WITH(openssl-digests, [  --with-openssl-digests  Use openssl digests (default=yes)], enable_openssl_digests="$withval", enable_openssl_digests="yes") 
+AC_ARG_WITH(openssl-digests, [  --with-openssl-digests  Use openssl digests (default=yes)], enable_openssl_digests="$withval", enable_openssl_digests="no") 
 
 # This is needed because autoheader can't work out which computed
 # symbols must be pulled from acconfig.h into config.h.in
@@ -658,7 +658,7 @@ if test "$have_libdb" = yes; then
   storages_available="$storages_available hashes(bdb $bdb_version)"
 fi
 
-AC_ARG_WITH(threestore, [  --with-threestore(=CONFIG|yes|no)  Location of 3store-config (default=auto)], tstore_config="$withval", tstore_config="")
+AC_ARG_WITH(threestore, [  --with-threestore(=CONFIG|yes|no)  Location of 3store-config (default=no)], tstore_config="$withval", tstore_config="no")
 if test "X$tstore_config" != "Xno"; then
   if test "X$tstore_config" != "Xyes" ; then
     TSTORE_CONFIG=$tstore_config
@@ -682,7 +682,7 @@ else
 fi
 
 
-AC_ARG_WITH(mysql, [  --with-mysql(=CONFIG|yes|no) Location of MySQL 3/4 mysql_config (default=auto)], mysql_config="$withval", mysql_config="")
+AC_ARG_WITH(mysql, [  --with-mysql(=CONFIG|yes|no) Location of MySQL 3/4 mysql_config (default=no)], mysql_config="$withval", mysql_config="no")
 if test "X$mysql_config" != "Xno"; then
   if test "X$mysql_config" != "Xyes" ; then
     MYSQL_CONFIG=$mysql_config
@@ -734,8 +734,8 @@ fi
 
 
 
-AC_ARG_WITH(sqlite, [  --with-sqlite(=yes|no|2|3) Enable SQLite store with version (default=auto)], sqlite_api_version="$withval", sqlite_api_version="auto")
-AC_ARG_WITH(sqlite-version, [  --with-sqlite-version(=yes|no|2|3) SQLite version to use (default=auto)], sqlite_api_version="$withval")
+AC_ARG_WITH(sqlite, [  --with-sqlite(=yes|no|2|3) Enable SQLite store with version (default=no)], sqlite_api_version="$withval", sqlite_api_version="no")
+AC_ARG_WITH(sqlite-version, [  --with-sqlite-version(=yes|no|2|3) SQLite version to use (default=no)], sqlite_api_version="no")
 if test "x$sqlite_api_version" = "xyes" ; then
   sqlite_api_version="auto"
 fi
@@ -785,7 +785,7 @@ else
   AC_MSG_RESULT(no)
 fi
 
-AC_ARG_WITH(postgresql, [  --with-postgresql(=CONFIG|yes|no)  Location of pg_config (default=auto)], pg_config="$withval", postgresql_config="")
+AC_ARG_WITH(postgresql, [  --with-postgresql(=CONFIG|yes|no)  Location of pg_config (default=no)], pg_config="$withval", pg_config="no")
 if test "X$pg_config" != "Xno"; then
   if test "X$pg_config" != "Xyes" ; then
     PG_CONFIG=$pg_config
diff --git a/redland/raptor/configure.ac b/redland/raptor/configure.ac
index 55878ed..8c45d5a 100644
--- a/redland/raptor/configure.ac
+++ b/redland/raptor/configure.ac
@@ -26,7 +26,7 @@ dnl
 AC_INIT(Raptor RDF Parser, 1.4.18, http://bugs.librdf.org/, raptor)
 AC_PREREQ(2.50)
 AC_CONFIG_SRCDIR(src/raptor_general.c)
-AC_REVISION($Revision: 1.1 $)
+AC_REVISION($Revision: 1.1 $)
 
 libxml_min_version=2.6.8
 
@@ -683,7 +683,7 @@ CPPFLAGS="$oCPPFLAGS"
 LIBS="$oLIBS"
 
 
-AC_ARG_WITH(xml-parser, [  --with-xml-parser=NAME    Use XML parser - libxml (default), expat], xml_parser="$withval", xml_parser="libxml") 
+AC_ARG_WITH(xml-parser, [  --with-xml-parser=NAME    Use XML parser - libxml (default), expat], xml_parser="$withval", xml_parser="expat") 
 
 for xml_parser_name in $xml_parser libxml expat; do
   case $xml_parser_name in
@@ -790,7 +790,7 @@ if test "x" = "y"; then
 fi
 
 AC_MSG_CHECKING(RDF parsers required)
-AC_ARG_ENABLE(parsers, [  --enable-parsers=LIST      Use RDF parsers (default=all)], parsers="$enableval") 
+AC_ARG_ENABLE(parsers, [  --enable-parsers=LIST      Use RDF parsers (default=all)], parsers="$enableval", parsers="rdfxml turtle") 
 if test "X$parsers" = Xall -o "X$parsers" = X; then
   parsers="$rdf_parsers_available"
   AC_MSG_RESULT(all)
@@ -875,7 +875,7 @@ if test "x" = "y"; then
 fi
 
 AC_MSG_CHECKING(RDF serializers required)
-AC_ARG_ENABLE(serializers, [  --enable-serializers=LIST  Use RDF serializers (default=all)], serializers="$enableval") 
+AC_ARG_ENABLE(serializers, [  --enable-serializers=LIST  Use RDF serializers (default=all)], serializers="$enableval", serializers="ntriples") 
 if test "X$serializers" = Xall -o "X$serializers" = X; then
   serializers="$rdf_serializers_available"
   AC_MSG_RESULT(all)
@@ -962,7 +962,7 @@ if test $ac_cv_header_fetch_h = yes; then
 fi
 
 
-AC_ARG_WITH(www, [  --with-www=NAME           Use WWW library - curl (default), xml, libfetch, none], www="$withval", www="curl") 
+AC_ARG_WITH(www, [  --with-www=NAME           Use WWW library - curl, xml, libfetch, none (default)], www="$withval", www="none") 
 
 for www_name in $www curl xml libfetch none; do
   case $www_name in
diff --git a/redland/rasqal/configure.ac b/redland/rasqal/configure.ac
index 81332a8..61aad50 100644
--- a/redland/rasqal/configure.ac
+++ b/redland/rasqal/configure.ac
@@ -25,7 +25,7 @@ dnl
 AC_INIT(Rasqal RDF Query Library, 0.9.16, http://bugs.librdf.org/, rasqal)
 AC_PREREQ(2.50)
 AC_CONFIG_SRCDIR(src/rasqal_general.c)
-AC_REVISION($Revision: 1.1 $)
+AC_REVISION($Revision: 1.1 $)
 
 AM_INIT_AUTOMAKE([1.7 check-news std-options -Wall])
 AM_CONFIG_HEADER(src/rasqal_config.h)
@@ -276,7 +276,7 @@ AC_TRY_LINK([#ifdef HAVE_GETOPT_H
 
 # raptor is REQUIRED despite the checking here
 AC_CHECK_PROG(RAPTOR_CONFIG, raptor-config, raptor-config)
-AC_ARG_WITH(raptor, [  --with-raptor=SOURCE        Pick raptor source - system/internal (default auto)], with_raptor="$withval", with_raptor="auto")
+AC_ARG_WITH(raptor, [  --with-raptor=SOURCE        Pick raptor source - system/internal (default internal)], with_raptor="$withval", with_raptor="internal")
 
 RAPTOR_MIN_VERSION=1.4.17
 AC_SUBST(RAPTOR_MIN_VERSION)
@@ -623,7 +623,7 @@ if test "x" = "y"; then
 fi
 
 AC_MSG_CHECKING(query languages required)
-AC_ARG_ENABLE(query-languages, [  --enable-query-languages=LIST   Enable query languages (default=all)], query_languages="$enableval") 
+AC_ARG_ENABLE(query-languages, [  --enable-query-languages=LIST   Enable query languages (default=all)], query_languages="$enableval", query_languages="sparql") 
 if test "X$query_languages" = Xall -o "X$query_languages" = X; then
   query_languages="$rdf_query_languages_available"
   AC_MSG_RESULT(all)
@@ -741,7 +741,7 @@ fi
 need_mpfr=0
 need_gmp=0
 
-AC_ARG_WITH(decimal, [  --with-decimal=NAME           Use decimal library - mpfr (default), gmp, none], decimal="$withval", decimal="mpfr") 
+AC_ARG_WITH(decimal, [  --with-decimal=NAME           Use decimal library - mpfr, gmp, none (default)], decimal="$withval", decimal="none") 
 for decimal_name in $decimal mpfr gmp none; do
   case $decimal_name in
     mpfr)
