Index: configure.ac
===================================================================
--- configure.ac	(revision 12937)
+++ configure.ac	(working copy)
@@ -1,28 +1,24 @@
 # Copyright (C) 2002-2008 Erik de Castro Lopo (erikd AT mega-nerd DOT com).
 
 dnl Require autoconf version >= 2.57)
-AC_PREREQ(2.57)
+AC_PREREQ([2.57])
 
 AC_INIT([libsamplerate],[0.1.7],[erikd@mega-nerd.com])
 
 # Put config stuff in Cfg.
-AC_CONFIG_AUX_DIR(Cfg)
+AC_CONFIG_AUX_DIR([Cfg])
 
 AC_CONFIG_SRCDIR([src/samplerate.c])
 AC_CANONICAL_TARGET([])
 
 AC_CONFIG_MACRO_DIR([M4])
 
-AM_INIT_AUTOMAKE($PACKAGE_NAME,$PACKAGE_VERSION)
+AM_INIT_AUTOMAKE
 dnl Audacity policy: don't enable automatic rebuild of configure et al if 
 dnl sources change
 AM_MAINTAINER_MODE([disable])
 AC_CONFIG_HEADER(src/config.h)
 
-
-dnl Add parameters for aclocal
-AC_SUBST(ACLOCAL_AMFLAGS, "-I M4")
-
 AC_LANG([C])
 
 #------------------------------------------------------------------------------------
Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12937)
+++ Makefile.am	(working copy)
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I M4
+
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 
 SUBDIRS = src doc examples tests
Index: examples/Makefile.am
===================================================================
--- examples/Makefile.am	(revision 12937)
+++ examples/Makefile.am	(working copy)
@@ -6,7 +6,7 @@
 noinst_PROGRAMS = varispeed-play timewarp-file
 
 SAMPLERATEDIR =../src
-INCLUDES = -I$(srcdir)/$(SAMPLERATEDIR) @OS_SPECIFIC_INCLUDES@
+AM_CPPFLAGS = -I$(srcdir)/$(SAMPLERATEDIR) @OS_SPECIFIC_INCLUDES@
 
 OS_SPECIFIC_LINKS = @OS_SPECIFIC_LINKS@
 
Index: tests/Makefile.am
===================================================================
--- tests/Makefile.am	(revision 12937)
+++ tests/Makefile.am	(working copy)
@@ -5,7 +5,7 @@
 					downsample_test
 
 SAMPLRATEDIR =../src
-INCLUDES = -I$(srcdir)/$(SAMPLRATEDIR)
+AM_CPPFLAGS = -I$(srcdir)/$(SAMPLRATEDIR)
 noinst_HEADERS = util.h
 
 #===============================================================================
