Index: lib-src/libvamp/Makefile.in
===================================================================
--- lib-src/libvamp/Makefile.in	(revision 13509)
+++ lib-src/libvamp/Makefile.in	(working copy)
@@ -3,6 +3,9 @@
 # libraries, example plugins, and the test host.  Please adjust to
 # suit your operating system requirements.
 
+srcdir      = @srcdir@
+VPATH       = @srcdir@
+
 APIDIR		= vamp
 
 SDKDIR		= vamp-sdk
@@ -40,7 +43,7 @@
 #
 CC		= @CC@
 CXX		= @CXX@
-CXXFLAGS	= -I. @CXXFLAGS@ @SNDFILE_CFLAGS@
+CXXFLAGS	= -I. -I$(srcdir) @CXXFLAGS@ @SNDFILE_CFLAGS@
 
 # ar, ranlib
 #
@@ -104,7 +107,7 @@
 # Additional flags for making a plugin.  This version script tells the
 # GNU linker to make all symbols in the library hidden except for the
 # public entry point.  It's not essential, but makes a tidier library.
-PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS) -Wl,--version-script=build/vamp-plugin.map
+PLUGIN_LDFLAGS		= $(DYNAMIC_LDFLAGS) -Wl,--version-script=${srcdir}/build/vamp-plugin.map
 
 
 ## For OS/X with g++:
@@ -218,8 +221,10 @@
 
 RDFGEN_TARGET	= \
 		$(RDFGENDIR)/vamp-rdf-template-generator
+prep:
+		mkdir -p $(SDKSRCDIR) $(HOSTSDKSRCDIR) $(EXAMPLEDIR)
 
-sdk:		sdkstatic $(SDK_DYNAMIC) $(HOSTSDK_DYNAMIC)
+sdk:		prep sdkstatic $(SDK_DYNAMIC) $(HOSTSDK_DYNAMIC)
 
 sdkstatic:	$(SDK_STATIC) $(HOSTSDK_STATIC)
 		$(RANLIB) $(SDK_STATIC)
