Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12902)
+++ Makefile.am	(working copy)
@@ -32,15 +32,38 @@
 # files that only really viscous cleans remove
 MAINTAINERCLEANFILES = Makefile.in
 
-# other distributed files in case users don't have a complete set of auto*
-AUX_DIST = $(ac_aux_dir)/config.guess \
-   $(ac_aux_dir)/config.sub \
-   $(ac_aux_dir)/install-sh \
-   $(ac_aux_dir)/mdate-sh \
-   $(ac_aux_dir)/missing \
-   $(ac_aux_dir)/mkinstalldirs \
-   $(MAINTAINERCLEANFILES)
-
 # Documentation files that should be both distributed and installed in the doc
 # directory,
 dist_doc_DATA = README.txt license.txt
+
+EXTRA_DIST = \
+	allegro.htm \
+	allegroserial.cpp \
+	apps/allegroconvert.cpp \
+	apps/allegroconvert.vcproj \
+	apps/allegroplay.cpp \
+	apps/midicode.h \
+	apps/seq2midi.cpp \
+	apps/seq2midi.h \
+	changelog.txt \
+	notes.txt \
+	portsmf-VC8.sln \
+	portsmf-VC8.vcproj \
+	portsmf.sln \
+	portsmf.suo \
+	portsmf.vcproj \
+	portsmf.xcodeproj \
+	portsmf.xcodeproj/project.pbxproj \
+	portsmf.xcodeproj/rbd.mode1 \
+	portsmf_test \
+	portsmf_test/Makefile.osx \
+	portsmf_test/portsmf_test-VC8.vcproj \
+	portsmf_test/portsmf_test.cpp \
+	portsmf_test/portsmf_test.vcproj \
+	portsmf_test/scale.gro \
+	portsmf_test/test.gro \
+	portsmf_test/test2.gro \
+	portsmf_test/test5.gro \
+	todo.txt \
+	trace.cpp \
+	$(NULL)
