Index: Makefile.am
===================================================================
--- Makefile.am	(revision 12833)
+++ Makefile.am	(working copy)
@@ -1,5 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
+ACLOCAL_AMFLAGS = -I m4
 
 #AUTOMAKE_OPTIONS = foreign 1.6 dist-zip
 AUTOMAKE_OPTIONS = foreign 1.11 dist-zip dist-xz
Index: configure.in
===================================================================
--- configure.in	(revision 12833)
+++ configure.in	(working copy)
@@ -2,6 +2,7 @@
 
 AC_INIT([libogg],[1.3.1],[ogg-dev@xiph.org])
 
+AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR(src/framing.c)
 
 AM_INIT_AUTOMAKE
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(revision 12833)
+++ src/Makefile.am	(working copy)
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_builddir)/include
 
 lib_LTLIBRARIES = libogg.la
 
